src/functions/Output.ps1

```diff
@@ -64,6 +64,111 @@ $script:ReportTheme = DATA {
     }
 }
 
+function Write-PesterHostMessage {
+    param(
+        [Parameter(Position = 0, ValueFromPipeline = $true)]
+        [Alias('Message', 'Msg')]
+        $Object,
+
+        [System.ConsoleColor]
+        $ForegroundColor,
+
+        [System.ConsoleColor]
+        $BackgroundColor,
+
+        [switch]
+        $NoNewLine,
+
+        $Separator = ' ',
+
+        [ValidateSet('Ansi', 'Legacy', 'Plaintext')]
+        [string]
+        $RenderMode = $PesterPreference.Output.RenderMode.Value
+    )
+
+    begin {
```

Comment:
I am not using begin often enough, this will run on start of every call to the command right? Maybe it is worth moving this outside of the function so we don't re-create the hashtable for every write to output?

You could also store the color codes in Dictionary<ConsoleColor, string>, which is probably a bit faster than keying by string, or you could put the colors in string array, and index into that based on the enum value.

Reply:
> I am not using begin often enough, this will run on start of every call to the command right? 

Correct, once even if multiple items through pipeline. 

> Maybe it is worth moving this outside of the function so we don't re-create the hashtable for every write to output?

I originally did that, but didn't measure any notable difference (last year). Can test again, including some different collections. 👍

Reply:
We save a total of 20ms on 1000 runs by having it pre-defined. Hashtable with string-keys vs consolecolor-key vs compiled in c# (`readonly static Dictionary<consolecolor,string>`) have no impact (+/- 2ms). Tested in 5.1 and 7.2

I'd stick with this for now. I can clean it up with ConsoleColor-keys to avoid the `ToString()`-call

Reply:
> Since there's no real perf benefit, you can decide based on code style.

Keep it as is please, since there is no benefit to get from putting it outside of the function.

> I'll switch to ConsoleColor-keys just to avoid the ToString()-call
👍